### PR TITLE
Porting checkbox to Lit

### PR DESCRIPTION
**Pull request by dlockhart**
Checkbox time!

### CODE REVIEW

File: components/inputs/input-checkbox.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		));
+	}
+
+	/**
+	 * This is needed only for IE11 and Edge AND going from indeterminate to checked/unchecked.
+	 * When the indeterminate state is set, and the checkbox is clicked, the _handleChange
+	 * function is NOT triggered, therefore we have to detect the click and handle it ourselves.
+	 */
+	_handleClick() {
+		/*const browserType = window.navigator.userAgent;
```
Comment (dbatiste) [inline components/inputs/input-checkbox.js]: Are we keeping this code?  (same with the focus handler below)

File: components/inputs/test/input-checkbox.visual-diff.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	[
+		'checked',
+		'unchecked',
+		'indeterminate',
+		'multiline',
+		'hidden-label'
+	].forEach((name) => {
+		it(`${name}-focus`, async function() {
+			await page.evaluate((name) => document.querySelector(`#${name}`).focus(), name);
```
Comment (dbatiste) [inline components/inputs/test/input-checkbox.visual-diff.js]: Maybe 6 or 1/2 dozen of the other...
```
await page.$eval(`#${name}`, (elem) => elem.focus());
```

File: components/inputs/README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ There are various input components available:
 
 - - [Text](#text-inputs)
 - - [Search](#search-inputs)
+- - [Checkboxes](#checkboxes)
```
Comment (dbatiste) [inline components/inputs/README.md]: Looks like we're getting double bullets.  I'm guessing that's not intentional.

**Comment by dbatiste** [comment]
Looks like we could use a link to the inputs readme in the root readme.

File: components/inputs/test/input-checkbox.visual-diff.js (hunk 2 above)
Comment (dlockhart) [reply components/inputs/test/input-checkbox.visual-diff.js]: Sorry I must be missing something, where does `elem` come from in that example?

File: components/inputs/input-checkbox.js (hunk 1 above)
Comment (dlockhart) [reply components/inputs/input-checkbox.js]: Haha nope. I meant to test it out in IE11 and Edge and see what's up and forgot -- will fix.

File: components/inputs/input-checkbox.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+	static get properties() {
+		return {
+			ariaLabel: { type: String, attribute: 'aria-label' },
+			checked: { type: Boolean },
+			disabled: { type: Boolean },
+			indeterminate: { type: Boolean },
+			name: { type: String },
+			notTabbable: { type: Boolean, attribute: 'not-tabbable' },
+			value: { type: String }
```
Comment (dlockhart) [inline components/inputs/input-checkbox.js]: In the old Polymer code, pretty much everything was reflecting back to the attribute. I removed that... which in theory could break something, but I feel it's doubtful and kind of silly to always-be-reflectin'. Our buttons suffered from the same thing unfortunately.

File: components/inputs/input-checkbox.js (hunk 1 above)
Comment (dlockhart) [reply components/inputs/input-checkbox.js]: This has been cleaned up. The `_handleClick` stuff was needed for indeterminate business, but the focus stuff is no longer required.

File: components/inputs/input-checkbox.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+					name="${ifDefined(this.name)}"
+					tabindex="${ifDefined(tabindex)}"
+					type="checkbox"
+					.value="${this.value}"><span class="${classMap(textClasses)}"><slot></slot></span>
+			</label>
+		`;
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('input.d2l-input-checkbox');
```
Comment (dlockhart) [inline components/inputs/input-checkbox.js]: Something to watch out for: I originally had this as `querySelector('.d2l-input-checkbox')`, but that was failing in IE & Edge. The reason is that shady CSS hijacks the element name and uses that as a CSS class, so it was putting the `d2l-input-checkbox` class on the `label` element! Bad shady CSS.

File: components/inputs/test/input-checkbox.visual-diff.js (hunk 2 above)
Comment (dbatiste) [reply components/inputs/test/input-checkbox.visual-diff.js]: `$eval` is a [Puppeteer shorthand](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageevalselector-pagefunction-args-1) that accepts a selector, fetches the element, and passes it to the provided page function.  There is also a `$$eval` that will fetch an array that will be passed to the provided page function.

Comment (dbatiste) [reply components/inputs/test/input-checkbox.visual-diff.js]: So:
```
await page.$eval(selector, (elem) => { ... });
```
is the same as:
```
await page.evaluate((selector) => {
   const elem = document.querySelector(selector);
}, selector);
```

File: components/inputs/input-checkbox.js (hunk 5 above)
Comment (dbatiste) [reply components/inputs/input-checkbox.js]: I think I have been burned by that too.  :(

File: components/inputs/test/input-checkbox.visual-diff.js (hunk 2 above)
Comment (dlockhart) [reply components/inputs/test/input-checkbox.visual-diff.js]: Ah, cool! Will update.

**Pull request merged**